collect_app/src/main/java/org/odk/collect/android/instancemanagement/SubmitException.kt

```diff
@@ -4,8 +4,6 @@ import java.lang.Exception
 
 class SubmitException(val type: Type) : Exception() {
     enum class Type {
-        GOOGLE_ACCOUNT_NOT_SET,
-        GOOGLE_ACCOUNT_NOT_PERMITTED,
         NOTHING_TO_SUBMIT
```

Comment:
Should we maybe just kill `Type` here as there is now only one?

Reply:
Yeah we can do that but I've decided that since this pr contains a lot of changes I don't want to add any even small improvements to make reviewing easier and just remove what we don't need anymore. This is one of the examples but not the only one. But if you are ok with reviewing some additional commits I can fix this.

Reply:
Done.